services/src/atdd-staging/src/main/java/org/openkilda/atdd/staging/steps/FlowCrudSteps.java

```diff
@@ -146,31 +139,31 @@ public void defineFlowsOverActiveTraffgens() {
                 }
 
                 String flowId = format("%s-%s", srcSwitch.getName(), dstSwitch.getName());
-                builder.addFlow(flowId, srcSwitch, srcTraffGen.getSwitchPort(), dstSwitch, dstTraffGen.getSwitchPort());
+                flowSetBuilder.addFlow(flowId, srcSwitch, srcTraffGen.getSwitchPort(), dstSwitch,
+                        dstTraffGen.getSwitchPort());
             });
         });
-
-        flows = builder.getFlows();
     }
 
     @And("^each flow has unique flow_id$")
     public void setUniqueFlowIdToEachFlow() {
-        flows.forEach(flow -> flow.setId(format("%s-%s", flow.getId(), UUID.randomUUID().toString())));
+        flowSetBuilder.getFlows()
+                .forEach(flow -> flow.setId(format("%s-%s", flow.getId(), UUID.randomUUID().toString())));
```

Comment:
This is not obvious that ***Builder builds always the same object. Many think that a builder must return a new instance each time.